include/seqan3/search/algorithm/detail/search_scheme_algorithm.hpp

```diff
@@ -0,0 +1,600 @@
+    {
+        delegate(it);
+        return true;
+    }
+    // Exact search in current block.
+    else if ((max_error_left_in_block == 0) && (rb - lb - 1 != blocks_length[block_id]) ||
+             (error_left.total == 0 && min_error_left_in_block == 0))
+    {
+        if (search_ss_exact<abort_on_hit>(it, query, lb, rb, errors_spent, block_id, go_right, search, blocks_length,
+                                          error_left, delegate) && abort_on_hit)
+        {
+            return true;
+        }
+    }
+    // Approximate search in current block.
+    // i.e. blocks_length[block_id] - (rb - lb - (lb != rb)) >= min_error_left_in_block
+    else if (error_left.total > 0)
+    {
+        // Insertion
+        if (error_left.insertion > 0)
+        {
+            using size_type = typename iterator_t::size_type;
+
+            size_type const lb2 = lb - !go_right;
+            size_type const rb2 = rb + go_right;
+
+            search_param error_left2{error_left};
+            error_left2.total--;
+            error_left2.insertion--;
+            // At the end of the current block
+            if (rb - lb == blocks_length[block_id])
+            {
+                // Leave the possibility for one or multiple deletions at the end of a block.
+                // Thus do not change the direction (go_right) yet.
+                // TODO: benchmark the improvement on preventing insertions followed by a deletion and vice versa. Does
+                // it pay off the additional complexity and documentation for the user? (Note that the user might only
+                // allow for insertions and deletion and not for mismatches).
+                if (search_ss_deletion<abort_on_hit>(it, query, lb2, rb2, errors_spent + 1, block_id, go_right, search,
+                                                     blocks_length, error_left2, delegate) && abort_on_hit)
+                {
+                    return true;
+                }
+            }
+            else
+            {
+                if (search_ss<abort_on_hit>(it, query, lb2, rb2, errors_spent + 1, block_id, go_right, search,
+                                            blocks_length, error_left2, delegate) && abort_on_hit)
+                {
+                    return true;
+                }
+            }
+        }
+        if (search_ss_children<abort_on_hit>(it, query, lb, rb, errors_spent, block_id, go_right,
+                                             min_error_left_in_block, search, blocks_length, error_left, delegate) &&
+            abort_on_hit)
+        {
+            return true;
+        }
+    }
+    return false;
+}
+
+/*!\brief Searches a query sequence in a bidirectional index using search schemes.
+ *
+ * \details Reference:
+ *          Kianfar, K., Pockrandt, C., Torkamandi, B., Luo, H., & Reinert, K. (2018).
+ *          Optimum Search Schemes for Approximate String Matching Using Bidirectional FM-Index. bioRxiv, 301085.
```

Comment:
Don't reference this everywhere, only reference it once. Also don't put details before the tparam/param documentation.